src/creature_states_combt.c

```diff
@@ -1860,11 +1860,12 @@ CrInstance get_self_spell_casting(const struct Thing *thing)
     if (creature_would_benefit_from_healing(thing))
     {
         INSTANCE_RET_IF_AVAIL(thing, CrInst_HEAL);
+        // Ranged heal can be also casted on itself.
```

Comment:
I think it would be best if this is determined by if the instance has the SELF_BUFF property too or not.

Reply:
Done.

Reply:
With the latest version I changed the instance to:
```ini
[instance50]
Name = RANGED_HEAL
Time = 10
ActionTime = 6
ResetTime = 400
FPTime = 5
FPActionTime = 3
FPResetTime = 250
FPInstantCast = 0
ForceVisibility = 1
TooltipTextID = 1066
; @todo May need new sprites.
SymbolSprites = 414
Graphics = ATTACK
RangeMin = MIN
RangeMax = 5120
PrimaryTarget = 6
Properties = RANGED_BUFF
Function = creature_cast_spell SPELL_SPEED 0
ValidateFunc = validate_source_generic validate_target_generic
SearchTargetsFunc = search_target_generic
```
So without the SELF_BUFF, but it then still cast speed on itself.

Reply:
ValidateFunc and SearchTargetsFunc look strange. Why use generic ones?

Reply:
I thought that was where these functions where for. I tried using it to cast speed on other creatures, and it did work, but it also cast it on caster which I tried to avoid.

Reply:
I will address your concern in validate_target_generic() in the next patch.

Reply:
By next patch you mean what you did on this PR, or you want to implement this in a different PR after this one is complete?
Because currently when I try the other functions you implemented with just a RANGED_BUFF, it still targets the caster.

Reply:
I meant what I did on this PR. Are you sure that spell is launched by this function? The spell you used might be launched by process_creature_self_spell_casting. I can test it for you if you could tell me how your test is conducted.

Reply:
I changed instance50 in creature.cfg to the instance I posted 5 posts up.

Reply:
In the instance you posted, the ValidateFunc and SearchTargetsFunc are incorrect. I suppose you should just remove the SELF_BUFF property. Why did you change ValidateFunc/SearchTargetsFunc to the generic ones?
I cannot reproduce the behavior you described. I removed the SELF_BUFF from the instance50, I am sure that Druid won't heal itself.

Reply:
> In the instance you posted, the ValidateFunc and SearchTargetsFunc are incorrect. I suppose you should just remove the SELF_BUFF property. Why did you change ValidateFunc/SearchTargetsFunc to the generic ones? I cannot reproduce the behavior you described. I removed the SELF_BUFF from the instance50, I am sure that Druid won't heal itself.

I changed this too `Function = creature_cast_spell SPELL_SPEED 0`
So I switched it to a speed buff. I believed the other validate functions you made were for cases like that, so this is how I tested that.
As you see, this is what I have for properties, so no SELF_BUFF at all:
`Properties = RANGED_BUFF`

And he casts SPEED on himself, without a SELF_BUFF.
But if not this, how should those new 'generic' functions be used?

Reply:
> > In the instance you posted, the ValidateFunc and SearchTargetsFunc are incorrect. I suppose you should just remove the SELF_BUFF property. Why did you change ValidateFunc/SearchTargetsFunc to the generic ones? I cannot reproduce the behavior you described. I removed the SELF_BUFF from the instance50, I am sure that Druid won't heal itself.
> 
> I changed this too `Function = creature_cast_spell SPELL_SPEED 0` So I switched it to a speed buff. I believed the other validate functions you made were for cases like that, so this is how I tested that. As you see, this is what I have for properties, so no SELF_BUFF at all: `Properties = RANGED_BUFF`
> 
> And he casts SPEED on himself, without a SELF_BUFF. But if not this, how should those new 'generic' functions be used?

For SPEED, yes, the generic ones should be used, but the instance you posted is not SPEED. It is RANGED_HEAL.

Reply:
Sorry, I indeed forgot to make an extra shot and spell for it. The instance name should not matter at all but I changed it too. I believe I now fully configured it correctly and it still does not work. I send the files I used for testing on discord for you to check.

I do believe get_best_self_preservation_instance_to_use will cast this instance either. It does check for SELF_BUFF there on higher numbers.

Reply:
@Linvail I saw your commit and noticed yours does indeed work. I also found out why it did not work for me .
You can revert that commit with the temporary spell.